pkg/rebuild/maven/infer.go

```diff
@@ -34,121 +34,71 @@ import (
 	"github.com/go-git/go-git/v5/storage"
 	"github.com/google/oss-rebuild/internal/uri"
 	"github.com/google/oss-rebuild/pkg/rebuild/rebuild"
-	mavenreg "github.com/google/oss-rebuild/pkg/registry/maven"
+	"github.com/google/oss-rebuild/pkg/registry/maven"
 	"github.com/pkg/errors"
 )
 
-type RepoConfig struct {
-	*git.Repository
-	URI    string
-	Dir    string
-	RefMap map[string]string
-}
-
-type BuildConfig struct {
-	Repo string
-	Dir  string
-	Ref  string
-	// TODO: Switch to an interface that can represent different build types.
-	Build MavenBuild
-}
-
-type MavenBuild struct {
-	// JDKVersion is the version of the JDK to use for the build.
-	JDKVersion string
-}
-
-func getPomXML(tree *object.Tree, path string) (pomXML mavenreg.PomXML, err error) {
-	f, err := tree.File(path)
+func (Rebuilder) InferRepo(ctx context.Context, t rebuild.Target, mux rebuild.RegistryMux) (string, error) {
+	pom, err := NewPomXML(ctx, t, mux)
 	if err != nil {
-		return
+		return "", err
 	}
-	p, err := f.Contents()
-	if err != nil {
-		return
+	// In some projects the groupId only exists on the parent.
```

Comment:
hmm this is a surprising place for these edits. would it be possible to move down to InferStrategy? In fact, I'm surprised these have any side effects outside this function. Is there caching somewhere that stores these changes?